src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2023,6 +2023,52 @@ class NodeUtils(object):
                 gpus[gpuid]['gis'] = {}
             gpus[gpuid]['gis'][giid] = gi
 
+        # Find all MIG UUIDs using nvidia-smi -L
+        # and update it later while finding the CIs
+        uuid_list = []
+        cmd = [self.cfg['nvidia-smi'], '-L']
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'NVIDIA SMI command: %s' % cmd)
+        time_start = time.time()
+        out = []
+
+        try:
+            # Try running the nvidia-smi command
+            process = subprocess.Popen(cmd, shell=False,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE,
+                                       universal_newlines=True)
+            out = process.communicate()[0].split('\n')
+        except Exception:
+            pbs.logmsg(pbs.EVENT_DEBUG4, 'Failed to execute: %s' %
+                       " ".join(cmd))
+            pbs.logmsg(pbs.EVENT_DEBUG3, '%s: No MIGs found' % caller_name())
+            return
+        elapsed_time = time.time() - time_start
+        if elapsed_time > 2.0:
+            pbs.logmsg(pbs.EVENT_DEBUG,
+                       '%s: nvidia-smi call took %f seconds' %
+                       (caller_name(), elapsed_time))
+
+        # format is:
+        # GPU gpuID: gpu_name (UUID: GPU UUID)
+        # MIG xg.ygb      Device  0: (UUID: MIG-UUID)
+        # MIG xg.ygb      Device  1: (UUID: MIG-UUID)
+
+        line_count = 0
+        for out_line in out:
+            # Don't capture the first list element which has the GPU UUID and
+            # the last line, which has only a newline in it
+            if(line_count == 0 or line_count == len(out)-1):
+                line_count += 1
+                continue
+
+            else:
+                mig_uuid = out_line.split()[-1].rstrip(")")
+                uuid_list.append(mig_uuid)
```

Comment:
Instead of having a `uuid_list`, consider a `uuid_map` with a key of (gpu_id, device_id)